proxy/http_headers.go

```diff
@@ -58,7 +58,12 @@ func addHeaders(r *http.Request, cfg config.Proxy, stripPath string) error {
 	// http proxy which sets it.
 	ws := r.Header.Get("Upgrade") == "websocket"
 	if ws {
-		r.Header.Set("X-Forwarded-For", remoteIP)
+		targetHeader := []string{remoteIP}
+		sourceHeader := r.Header.Get("X-Forwarded-For")
+		if sourceHeader != "" {
```

Comment:
There are some subtle things going on here.  It would be good to mirror the functionality in the standard library:

https://github.com/golang/go/blob/master/src/net/http/httputil/reverseproxy.go#L296

The Get() call on the header only grabs the first element if there is more than one.